calendar-webadmin/src/main/java/com/linagora/calendar/webadmin/ResourceRoutes.java

```diff
@@ -156,84 +182,149 @@ private ResourceDTO getResource(Request req) {
         return resourceDAO.findById(id)
             .flatMap(this::toDto)
             .blockOptional()
-            .orElseGet(() -> {
-                throw ErrorResponder.builder()
-                    .statusCode(HttpStatus.NOT_FOUND_404)
-                    .type(ErrorResponder.ErrorType.NOT_FOUND)
-                    .message("Resource do not exist")
-                    .cause(new RuntimeException())
-                    .haltError();
-            });
+            .orElseThrow(() -> resourceNotFound(id));
     }
 
     private String deleteResource(Request req, Response res) {
         ResourceId id = new ResourceId(req.params("id"));
 
-        try {
-            resourceDAO.softDelete(id).block();
-            res.status(204);
-            return "";
-        } catch (ResourceNotFoundException e) {
-            throw ErrorResponder.builder()
-                .statusCode(HttpStatus.NOT_FOUND_404)
-                .type(ErrorResponder.ErrorType.NOT_FOUND)
-                .message("Resource do not exist")
-                .cause(new RuntimeException())
-                .haltError();
-        }
+        return findNotDeletedResource(id)
+            .flatMap(currentResource -> resolveAdminUsernames(currentResource)
+                .flatMap(adminUsers -> {
+                    if (!adminUsers.isEmpty()) {
```

Comment:
dav client already protects against noop updates. Please remove the check here, we do not need it.